Sources/ArgumentParser/Usage/HelpGenerator.swift

```diff
@@ -140,7 +141,8 @@ internal struct HelpGenerator {
     /// more elements at a time.
     var args = commandStack.argumentsForHelp(visibility: visibility)[...]
     while let arg = args.popFirst() {
-      guard arg.help.visibility == .default else { continue }
+      guard arg.help.visibility.isAtLeastAsVisible(as: visibility)
```

Comment:
Shouldn't `var args = commandStack.argumentsForHelp(visibility: visibility)[...]` make this unnecessary?

Reply:
I was thinking about this, I don't think removing it will work because the visibility in ArugmentSetProvider isn't correct for non-OptionGroups. But maybe I'm wrong

Reply:
I updated `ArgumentSet(_:visibility:)` so that it does the filtering, as you'd expect from the spelling.